security/pkg/stsservice/server/server.go

```diff
@@ -135,7 +136,10 @@ func (s *Server) validateStsRequest(req *http.Request) (security.StsRequestParam
 	}
 
 	if stsServerLog.DebugEnabled() {
-		reqDump, _ := httputil.DumpRequest(req, true)
+		debugReq := req
+		// Limit req body to 10 mb
+		debugReq.Body = io.NopCloser(io.LimitReader(req.Body, 1024*1024*10))
```

Comment:
the NopCloser seems problematic? I think we need to actually close it?

http.MaxBytesReader may be more appropriate

Reply:
sgtm, just checked `NopClosed` is depreceated in golang.